hack/update-codegen.sh

```diff
@@ -611,6 +588,10 @@ function codegen::applyconfigs() {
         done
     fi
 
+    git_find -z 'staging/src/k8s.io/client-go/**/*.go' \
+        | { xargs -0 grep -z -l "^// Code generated by applyconfiguration-gen. DO NOT EDIT.$" || true; } \
```

Comment:
I agree that grepping for comments is sort of awful, but I don't want to start open-coding exclusions.  These things should all be as metadata-driven as possible.

Here's an alternative: What if we renamed all of these output files?  Instead of `staging/src/k8s.io/client-go/applyconfigurations/apps/v1/deployment.go` maybe `staging/src/k8s.io/client-go/applyconfigurations/apps/v1/generated.deployment.go` or `deployment.generated.go` or `zz_generated.applyconfig.deployment.go` or something?  How much would people hate that?

Reply:
Indeed, grepping for "DO NOT EDIT" causes verify-boilerplate to flag these files as "generated", becuase it greps (indiscriminately) for "DO NOT EDIT" (so it matches the "grep" command in the scripts).  BLeeeeeech.

Reply:
2 ideas here:

1) We make sure EVERY generated file has a discoverable name.  I don't know what this would do for things like proto files which are generated (isn't filename important in proto?) but it would work for .go files.

2) We make sure EVERY generated file has a CLEARLY documented comment that is not easy to mistake.  E.g. `^[/*# ]+ THIS FILE IS AUTO-GENERATED: DO NOT EDIT$` and maybe another line like  `^[/*# ]+ +k8s:generated-by: (.*)$` where the match-group is the tool name for easier grepping.  Basically, write a spec for genfile detection.

3) rot13 the strings in shell scripts :)

Reply:
Either a fixed filename or a fixed generated token in the file (like the `+k8s:generated-by: ...` bit) we can match on seem reasonable. I'd probably support both approaches so we don't have to rename the world and can accommodate generated files where the filename is important/fixed for some other reason.

Reply:
Either approach is pretty enormous.  To stick such a tag in emitted files means touching gengo and we still have the problem that the script are grepping for a string which other tools will find in those scripts.

I'm going to see if there's a less dramatic answer if we just hold our noses a bit.  Really, the boilerplate script should just ignore diffs that are "THIS FILE IS AUTO-GENERATED".

Reply:
I think I can make boilerplate.py smarter.

Reply:
> we still have the problem that the script are grepping for a string which other tools will find in those scripts

excluding non-go files or the hack dir seems like it would generally work, no?

> I think I can make boilerplate.py smarter.

that works as well